torchrl/modules/models/utils.py

```diff
@@ -103,3 +104,24 @@ def _find_depth(depth: Optional[int], *list_or_ints: Sequence):
             f"num_cells) to be a a list or tuple. Got {tuple(type(item) for item in list_or_ints)}"
         )
     return depth
+
+
+def create_on_device(
+    module_class: Type[nn.Module], device: DEVICE_TYPING, *args, **kwargs
+) -> nn.Module:
+    """
+    Create a new instance of `module_class` on `device`.
+
+    The new instance is created directly on the device if its constructor supports this.
+
+    Args:
+        module_class (Type[nn.Module]): the class of module to be created.
+        device (DEVICE_TYPING): device to create the module on.
+        *args: positional arguments to be passed to the module constructor.
+        **kwargs: keyword arguments to be passed to the module constructor.
+    """
+    try:
```

Comment:
try / except are a bit expensive
Even if we don't create modules very often in training loops, we do in the tests, and on the long run this risks to have a negative effect on our test speed.
Do you think we can find some other way of doing this?